application/src/ext-test/java/org/opentripplanner/ext/fares/impl/gtfs/CombinedInterlinedLegsFareServiceTest.java

```diff
@@ -80,7 +80,7 @@ void modes(CombinationMode mode, Itinerary itinerary, Money totalPrice, String h
       .values()
       .stream()
       .distinct()
-      .map(p -> p.product().price())
+      .map(p -> p.product().fareProduct().price())
```

Comment:
Nit: `<product>.product().fareProduct()` doesn't sit well because of the attribute drilling and the repeated use of "product"... Is that because `p.product()` is of type `FareOffer`, in which case `.product()` could be renamed to `.offer()` for instance? Or, you could create a helper method to access the `.price()` from each leg product.